### PR TITLE
fixing for compat

### PR DESCRIPTION
**Pull request by iquah1**
Currently relies on PALISADE. This PR makes it use OpenFHE

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -1,28 +1,35 @@
...
+    set(CMAKE_EXE_LINKER_FLAGS ${OpenFHE_EXE_LINKER_FLAGS})
+    link_libraries(${OpenFHE_SHARED_LIBRARIES})
+endif ()
 
 set(CMAKE_RUNTIME_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/bin)
 
 # ADD src
 add_subdirectory( src )
 add_subdirectory( data )
+message(STATUS "NDEBUG: ${NDEBUG}")
```
Comment (mctriplett) [inline CMakeLists.txt]: Might be able to drop this

File: Contributing.md [hunk 2]
```diff
@@ -1,184 +1,178 @@
- # Contributing to PALISADE
+# Contributing to OpenFHE
 
-We are using git for version management and control. We use gitlab for
-issue and milestone tracking.
+We are using git for version management and control. We use gitlab for issue and milestone tracking.
```
Comment (mctriplett) [inline Contributing.md]: GitLab -> GitHub?

File: Contributing.md [hunk 3]
```diff
@@ -1,184 +1,178 @@
...
+deprecate large amounts of existing code. This way you can make sure your additions will be consistent with the planned
+release schedule of OpenFHE. It will also ensure that you base your changes on the most recent version of the
+development library.
 
-In addition to the workflow for Minor contributions the following is the
-requested procedure or a Major change.
+In addition to the workflow for Minor contributions the following is the requested procedure or a Major change.
 
-* Fork the `palisade-development` repository on GitLab
+* Fork the `openfhe-development` repository on GitLab
```
Comment (mctriplett) [inline Contributing.md]: GitLab -> GitHub

File: src/strsearch_enc_1.cpp [hunk 4]
```diff
@@ -1,46 +1,45 @@
 /* PALISADE C++ program implements the Rabin-Karp method for string
```
Comment (mctriplett) [inline src/strsearch_enc_1.cpp]: PALISADE -> OpenFHE

File: src/strsearch_enc_2.cpp [hunk 5]
```diff
@@ -403,9 +408,9 @@ int main()
   vecInt foundloc(0);
   for (size_t i = 0; i < vecResult.size(); i++) {
     auto unpackedVal = vecResult[i]->GetPackedValue();
-    for (size_t j = 0; j< ringsize; j++) {
-      if (unpackedVal[j] == 0) {
-      auto loc = i + offset[j];
+    for (size_t jj = 0; jj< ringsize; jj++) {
```
Comment (mctriplett) [inline src/strsearch_enc_2.cpp:411]: jj vs j, any particular reason for this?

Comment (iquah1) [reply src/strsearch_enc_2.cpp:411]: Shadowed a variable higher up. My compiler was giving me some issues

**Review by mctriplett — APPROVED** [review]

**Pull request merged**